### ISSUE TITLE
Optionally output installation process in real time

### ISSUE THREAD

**Issue by avishnyakov** [shurick81/Dynamics365Configuration#19]
Install process takes around 30-70 minutes depending on the resources. It would be nice to pull installation logs in real time. 

One trick which can be useful is below. It can be adapted into the install process to monitor it in real time.
```
# pull logs in the real time
Get-Content $path -Wait
``

Install-Dynamics365Server does the pulling of the job status, it could potentially read $path, compare new lines added, and the push int the console output in -verbose mode or with special flag.

Unattended installs would surely benefit from this feature.

**Comment by avishnyakov** [comment]
A very terrible example of how it might work together with https://github.com/shurick81/Dynamics365Configuration/blob/72120aba483e4705cc07f7ca1211af2cf16f6357/src/Dynamics365Configuration/Install-Dynamics365Server.psm1#L301

* calculate current/new lines
* get the diff
* colour it up
* pull again until done, that's already in Install-Dynamics365Server waiting loop

```powershell

$path = "C:\Users\_CRMAD~1\AppData\Local\Temp\CRMInstallationLog_2019-02-21-21-01-20Z.txt"

$lastLinesCount = 0


do {
    
    $linesCount = (Get-Content $path | Measure-Object -Line).Lines
    Write-Host $linesCount
    $newLinesCount = $linesCount - $lastLinesCount

    Write-Host "count    : $lastLinesCount"
    Write-Host "new lines: $newLinesCount"
    Write-Host ".."

    if($newLinesCount -gt 0) {
        $lines = get-content $path  | select -first ($newLinesCount - 1) -skip $lastLinesCount

        
        foreach($line in $lines) {

            $color = "Green"

            if($line.Contains("Verbose") -eq $True) {
                $color = "Cyan"
            }

            if($line.Contains("Warning") -eq $True) {
                $color = "Yellow"
            }

            if($line.Contains("Error") -eq $True) {
                $color = "Red"
            }

            Write-Host $line -Fore $color
        }
     } else {
           Write-Host "No new events"
     }

    $lastLinesCount = $linesCount

    Start-Sleep 3

} while ($True)

```

Output

```
..
7082
count    : 7076
new lines: 6
..
21:27:15|   Info| Executing Install action: Microsoft.Crm.Tools.Admin.NavigationPropertyNameDataPopulationAction
21:27:17|   Info| CrmAction execution time; NavigationPropertyNameDataPopulationAction; 00:00:01.5008199
21:27:17|   Info| Executing Install action: Microsoft.Crm.Tools.Admin.InstallDatabaseUpdatesAction
21:27:17|Verbose| Retrieving database base language code Data Source=CRM;Initial Catalog=uplift_MSCRM;Integrated Security=True;Enlist=False;Connect Timeout=150;Application Name=MSCRMMicrosof
t.Crm.Setup.Server;Workstation ID=CRM.Microsoft.Crm.Setup.Server ...
21:27:17|   Info| Database base language code: 1033.
7082
count    : 7082
new lines: 0
..
7082
count    : 7082
new lines: 0
..
7082
count    : 7102
new lines: 31
..
21:27:27|   Info| If required, Disable LabelCacheSharing in OrganizationLanguagePack table
21:27:27|Verbose| DatabaseAction execution time; Microsoft.Crm.SE.DatabaseActions.dll:FlipLabelCacheSharingBit; 00:00:00.0312251
21:27:27|   Info| Executing: action Bin\Microsoft.Crm.SE.DatabaseActions.dll:Execute with id a5697568-1f3b-409e-b5d2-a8dbb5571ca5 of type DllMethod to Update ImportData, ImportFile, ImportLo
g to be non-solutionaware.
21:27:27|   Info| Update ImportData, ImportFile, ImportLog to be non-solutionaware
21:27:46|Verbose| DatabaseAction execution time; Bin\Microsoft.Crm.SE.DatabaseActions.dll:Execute; 00:00:19.0000076
21:27:46|   Info| Executing: action DBUpdate_675671.sql with id 1efd0e5f-1b18-11e7-82bb-a0481c94431d of type SqlScript to Updates DependendentComponentId column type to guid from nvarchar in
 RuntimeDependencyBase table.
21:27:46|   Info| Executing: action DBUpdate_777780.sql with id e42c6ce9-b75d-46ae-a6e3-e602570e9ac2 of type SqlScript to Updates the child attributes of regardingobjectid to be part of acti
ve solution.
21:27:46|   Info| Updates the child attributes of regardingobjectid to be part of active solution
21:27:47|   Info| Did not detect single user -- not clearing pool
21:27:47|Verbose| DatabaseAction execution time; DBUpdate_777780.sql; 00:00:01.0156896
21:27:47|   Info| Executing: action DBUpdate_RelationshipExtraCondition.sql with id 93a7d6a6-4d80-4ba7-80bc-e8d0d0963acf of type SqlScript to Update relationshipextracondition properties.
21:27:47|   Info| Update relationshipextracondition properties
21:27:47|   Info| Did not detect single user -- not clearing pool
21:27:47|Verbose| DatabaseAction execution time; DBUpdate_RelationshipExtraCondition.sql; 00:00:00.4218216
21:27:47|   Info| Executing: action DeleteIncorrectChannelActivityOptions.sql with id 9559a9b6-9617-4a4f-97bc-5c46118821a9 of type SqlScript to Delete incorrect attributepicklistvalues.
21:27:47|   Info| Delete incorrect attributepicklistvalues
21:27:48|   Info| Did not detect single user -- not clearing pool
21:27:48|Verbose| DatabaseAction execution time; DeleteIncorrectChannelActivityOptions.sql; 00:00:00.6874787
21:27:48|   Info| Executing: action UpdateEntityRelationshipRole.sql with id 6364eca7-7365-48f5-891a-e09cfedcbb27 of type SqlScript to Update entityrelationshiprole properties.
21:27:48|   Info| Update entityrelationshiprole properties
21:27:48|   Info| Did not detect single user -- not clearing pool
21:27:48|Verbose| DatabaseAction execution time; UpdateEntityRelationshipRole.sql; 00:00:00.0781679
21:27:48|   Info| Executing: action DBUpdate_RibbonDiff_TabToCommandMap.sql with id 892d4e0d-f3bc-452a-80f9-74d4c5891470 of type SqlScript to Modifies the TabId fields on RibbonDiff and Ribb
onTabToCommandMap to make them nvarchar(200) instead of nvarchar(max) and add indexes to those fields.
21:27:48|   Info| Modifies the TabId fields on RibbonDiff and RibbonTabToCommandMap to make them nvarchar(200) instead of nvarchar(max) and add indexes to those fields
21:27:49|   Info| Did not detect single user -- not clearing pool
21:27:49|Verbose| DatabaseAction execution time; DBUpdate_RibbonDiff_TabToCommandMap.sql; 00:00:00.9062899
21:27:49|   Info| Executing: action DBUpdate_InsertMissingLabelsForOptionSetValues.sql with id e83c5925-c295-44f2-968e-0065f8fd6aa9 of type SqlScript to Insert missing labels for some option
s of the optionsets activitypointer_activitytypecode,mobileofflineenabledentities, sla_slaenabledentities..
21:27:49|   Info| Insert missing labels for some options of the optionsets activitypointer_activitytypecode,mobileofflineenabledentities, sla_slaenabledentities.
21:27:49|   Info| Did not detect single user -- not clearing pool
21:27:49|Verbose| DatabaseAction execution time; DBUpdate_InsertMissingLabelsForOptionSetValues.sql; 00:00:00.2187840
7133
```

**Comment by shurick81** [comment]
Thanks for the PR, I pushed the commits to a separate branch to continue from there

**Comment by shurick81** [comment]
Thanks, now this is finished and will be released in v1.2
